tests/test_jobs.py

```diff
@@ -0,0 +1,246 @@
+import pytest
+import requests
+from requests import HTTPError
+
+from dcos_test_utils.helpers import Url
+from dcos_test_utils.jobs import Jobs
+
+
+class MockResponse:
+    def __init__(self, json: dict, status_code: int):
+        self._json = json
+        self._status_code = status_code
+
+    def json(self):
+        return self._json
+
+    def raise_for_status(self):
+        """Throw an HTTPErrer based on status code."""
+        if self._status_code >= 400:
+            raise HTTPError('Throwing test error')
+
+    @property
+    def status_code(self):
+        return self._status_code
+
+
+class MockEmitter:
+    """Emulates a Session and responds with a queued response for each
+    request made. If no responses are queued, the request will raise
+    an error.
+
+    A history of requests are held in a request cache and can be
+    reviewed.
+    """
+
+    def __init__(self, mock_responses: list):
+        self._mock_responses = mock_responses
+        self._request_cache = list()
+
+    def request(self, *args, **kwargs):
+        self._request_cache.append((args, kwargs))
+        return self._mock_responses.pop(0)
+
+    def queue(self, response: list or MockResponse):
```

Comment:
Please make this only take a list. Its pretty trivial to wrap the single MockResponse in a list and will make the typing clear when this is used